lib/algora/admin.ex

```diff
@@ -16,6 +16,119 @@ defmodule Algora.Admin do
     Finch.build(:get, url) |> Finch.request(Algora.Finch)
   end
 
+  defp get_absolute_media_playlist(video) do
+    %ExM3U8.MediaPlaylist{timeline: timeline, info: info} = get_media_playlist(video, @tracks.video)
+
+    timeline = timeline
+    |> Enum.reduce([], fn x, acc ->
+        case x do
+          %ExM3U8.Tags.MediaInit{uri: uri} -> [
+            %ExM3U8.Tags.MediaInit{uri: Storage.to_absolute(:video, video.uuid, uri)}
+            | acc
+          ]
+          %ExM3U8.Tags.Segment{uri: uri, duration: duration} -> [
+            %ExM3U8.Tags.Segment{uri: Storage.to_absolute(:video, video.uuid, uri), duration: duration}
+            | acc
+          ]
+          others -> [others | acc]
+        end
+      end)
+    |> Enum.reverse()
+
+    %ExM3U8.MediaPlaylist{timeline: timeline, info: info}
+  end
+
+  defp merge_media_playlists(video, nil), do: get_absolute_media_playlist(video)
+
+  defp merge_media_playlists(video, playlist) do
+    new_playlist = video |> get_absolute_media_playlist
+
+    %ExM3U8.MediaPlaylist{
+      playlist
+      | timeline: playlist.timeline ++ [%ExM3U8.Tags.Discontinuity{} | new_playlist.timeline],
+      info: %ExM3U8.MediaPlaylist.Info{
+        playlist.info
+        | target_duration: max(playlist.info.target_duration, new_playlist.info.target_duration)
+      }
+    }
+  end
+
+  defp merge_playlists(videos) do
+    playlists = Enum.map(videos, &get_playlist/1)
+
+    streams = Enum.flat_map(playlists, &Map.get(&1, :items))
+    example_stream = Enum.at(streams, 1)
+
+    if Enum.all?(streams, fn x -> example_stream.resolution == x.resolution && example_stream.codecs == x.codecs end) do
+      max_bandwidth = Enum.map(streams, &Map.get(&1, :bandwidth)) |> Enum.max(&Ratio.gte?/2)
+      avg_bandwidth = streams
+        |> Enum.reduce(0, fn s, sum -> sum + s.average_bandwidth end)
+        |> then(&(&1 / Enum.count(streams)))
```

Comment:
This should be sum(s.average_bandwidth * s.segment_count) / sum(s.segment_count) I think

Reply:
I'm not sure I understood this.

Do you want it to be a weighted average depending of how many segments each stream has?

Reply:
Yeah exactly (see definition [here](https://github.com/membraneframework/membrane_http_adaptive_stream_plugin/blob/d032310cff947594889d74d77ac68ec232473a1b/lib/membrane_http_adaptive_stream/bandwidth_calculator.ex#L52)). Otherwise we're averaging over streams instead of segments

Idk if this actually matters but better to be safe than sorry. Some devices/browser/players can be really anal about such things